braindecode/preprocessing/eegprep_preprocess.py

```diff
@@ -0,0 +1,1139 @@
+"""Preprocessors using the EEGPrep package."""
+
+# Authors: Christian Kothe <christian.kothe@intheon.io>
+#
+# License: BSD-3
+
+import base64
+import json
+import logging
+import re
+from types import ModuleType
+from typing import Any, Sequence
+
+import numpy as np
+from mne import BaseEpochs
+from mne.io import BaseRaw
+
+from braindecode.preprocessing.preprocess import Preprocessor
+
+log = logging.getLogger(__name__)
+
+__all__ = [
+    "EEGPrep",
+    "RemoveDCOffset",
+    "Resampling",
+    "RemoveFlatChannels",
+    "RemoveDrifts",
+    "RemoveBadChannels",
+    "RemoveBadChannelsNoLocs",
+    "RemoveBursts",
+    "RemoveBadWindows",
+    "ReinterpolateRemovedChannels",
+    "RemoveCommonAverageReference",
+]
+
+try:
+    import eegprep
+except ImportError:
+
+    class EEGPrepMissing(ModuleType):
+        def __getattr__(self, _: str) -> Any:
+            raise ImportError(
+                "The eegprep package is required to use the EEGPrep preprocessor.\n"
+                "  Please install braindecode with the [eegprep] extra added as in\n"
+                "  'pip install braindecode[eegprep]' to use this functionality,\n"
+                "  or run 'pip install eegprep[eeglabio]' directly."
+            )
+
+    eegprep = EEGPrepMissing("eegprep")
```

Comment:
I would prefer something simpler than this.... but I can do in the last round of the revisions

Reply:
Simplified it to a minimal try/catch and a single check in the first place where eegprep is needed in the codebase. Note that `_soft_import()` doesn't do much other than setting it to `False` in the except case. It's arguably more customary to set it to `None,` but more importantly, it masks the import from code inspection tools and IDEs etc, so you don't get autocomplete, type checks, and so forth when working with functions/etc in `eegprep`.